### ISSUE TITLE
Problems when using brakeman as rake task

### ISSUE THREAD

**Issue by vinistock** [Shopify/bootsnap#54]
Hi, everyone. I'd just like to start saying that I find this gem pretty neat and that it does provide a significant performance improvement.

Now to the actual issue.

When adding bootsnap to a project and invoking rake (default task) or brakeman through the console, everything works fine.

On the other hand, if I create a rake task that includes both running brakeman and the specs, I receive and error. Here are the specific details

### Steps to reproduce

I am overwriting the default rake task to run brakeman:check (from brakeman) first and then run parallel:spec (from parallel_tests).

```ruby
Rake::Task['default'].clear

task :default do
  Rake::Task['brakeman:check'].invoke
  Rake::Task['parallel:spec'].invoke
end
```
### Expected behavior
The overwritten default rake task should run brakeman and then run the parallel specs.

### Actual behavior

I get numerous warnings and an error, as can be seen below.

```
TypeError: superclass mismatch for class Mapping
```

```
/HOME/USER/.rvm/gems/ruby-2.4.1@mygemset/gems/brakeman-3.6.2/bundle/ruby/2.3.0/gems/tilt-2.0.7/lib/tilt/mapping.rb:219: warning: already initialized constant Tilt::Mapping::LOCK
/HOME/USER/.rvm/gems/ruby-2.4.1@mygemset/gems/tilt-2.0.7/lib/tilt/mapping.rb:219: warning: previous definition of LOCK was here
/HOME/USER/.rvm/gems/ruby-2.4.1@mygemset/gems/brakeman-3.6.2/bundle/ruby/2.3.0/gems/tilt-2.0.7/lib/tilt/mapping.rb:263: warning: already initialized constant Tilt::Mapping::AUTOLOAD_IS_BROKEN
/HOME/USER/.rvm/gems/ruby-2.4.1@mygemset/gems/tilt-2.0.7/lib/tilt/mapping.rb:263: warning: previous definition of AUTOLOAD_IS_BROKEN was here
/HOME/USER/.rvm/gems/ruby-2.4.1@mygemset/gems/brakeman-3.6.2/bundle/ruby/2.3.0/gems/tilt-2.0.7/lib/tilt/template.rb:5: warning: already initialized constant Tilt::TOPOBJECT
/HOME/USER/.rvm/gems/ruby-2.4.1@mygemset/gems/tilt-2.0.7/lib/tilt/template.rb:5: warning: previous definition of TOPOBJECT was here
/HOME/USER/.rvm/gems/ruby-2.4.1@mygemset/gems/brakeman-3.6.2/bundle/ruby/2.3.0/gems/tilt-2.0.7/lib/tilt/template.rb:16: warning: already initialized constant Tilt::LOCK
/HOME/USER/.rvm/gems/ruby-2.4.1@mygemset/gems/tilt-2.0.7/lib/tilt/template.rb:16: warning: previous definition of LOCK was here
```

### System configuration
**Rails version**: 5.1.1
**Ruby version**: 2.4.1
**Brakeman version**: 3.6.2
**Parallel tests version**: 2.14.1
**OS**: Ubuntu 16.04

**Comment by jules2689** [comment]
This issue is similar to one [that happened on Bundler](https://github.com/bundler/bundler/issues/2068). Basically what happened in Bundler is that Psych ends up being double loaded, or reloaded actually.

What I can tell is happening here is that Tilt is also being double loaded. I think this may be a simple incompatibility as it seems that Brakeman loads Tilt, but so does your application - but does so at different locations due to brakeman having a bundled set of gems inside of it.

Installing brakeman locally, I can see that Brakeman literally bundles gems inside of its installation and pre-ships with gems. `Bootsnap` assumes certain paths remain stagnant and that we can point to specific paths using [path pre-scanning](https://github.com/Shopify/bootsnap#path-pre-scanning). This means that Brakeman loads a gem from location A while your app loads from B, which causes a redefinition and pre-scanning clash.

I am not certain the pre-scanning is the culprit, but I am certain that the bundled gems are suspect.

Will have to investigate this more with a minimal failing test case (likely with brakeman and parallel tests in a super simple ruby app).

**Comment by ojab** [comment]
Can you please check if https://github.com/Shopify/bootsnap/pull/136 fixes the issue? You can try it with `gem 'bootsnap', github: 'ojab/bootsnap', require: false` in Gemfile.

**Comment by vinistock** [comment]
@ojab I tried it with your version of bootstrap and I got this error when running rake:

```bash
cannot load such file -- ruby_parser
Please install the appropriate dependency: ruby_parser.
```

**Comment by vinistock** [comment]
@ojab Hey. I have tested again, after your pull requested was merged, using the bootsnap 1.3.0.beta2 version. It is now working and I'm yet to see any other issues. Thanks for looking into this, great work!

**Comment by burke** [comment]
Thanks everyone :tada:
